File: Backendd/ANIMO_Backend/src/main/java/edu/isep/animo_backend/models/Blimp.java
package edu.isep.animo_backend.models;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "tblBlimps")
public class Blimp {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private @Getter @Setter Long blimpid;
    private @Getter @Setter String name;
    private @Getter @Setter double longitude;
    private @Getter @Setter double latitude;

    @ManyToOne
    @JoinColumn(name = "farmid")
    private @Getter @Setter Farm farm;


}

File: Frontend/ANIMO/src/app/modules/shared/models/blimp.ts
import { Farm } from "./farm";

export class Blimp {
    blimpid: number;
    name: string;
    longitude: number;
    latitude: number;
    farm: Farm;
  
    constructor(blimpid: number, name: string, longitude: number, latitude: number, farm: Farm) {
      this.blimpid = blimpid;
      this.name = name;
      this.longitude = longitude;
      this.latitude = latitude;
        this.farm = farm;
    }
  }